core/primitives/src/shard_layout.rs

```diff
@@ -186,6 +425,52 @@ impl ShardLayout {
         )
     }
 
+    /// Returns the simple nightshade layout, version 4, that will be used in
+    /// production. It adds a new boundary account splitting the "game.hot.tg"
+    /// shard into two smaller shards. This is the first layout used in the
+    /// Instant Resharding and it is the first one where the shard id contiguity
+    /// is broken.
+    ///
+    /// TODO(resharding) Determine the shard layout for v4.
+    /// This layout is provisional, the actual shard layout should be determined
+    /// based on the fresh data before the resharding.
+    pub fn get_simple_nightshade_layout_v4() -> ShardLayout {
+        // the boundary accounts in lexicographical order
```

Comment:
nit: is it worth to add `todo!()` or similar compiler help?

Reply:
Add todo where? 

I'm hoping we can get v4 into nightly soon enough and start testing it there. I wouldn't want compiler panicking there.

Reply:
Just to prevent using the provisional layout by mistake, but anyway we are going to check this again before releasing